src/server/pps/server/api_server.go

```diff
@@ -2548,9 +2548,22 @@ func (a *apiServer) deletePipeline(pachClient *client.APIClient, request *pps.De
 		if err := a.authorizePipelineOp(pachClient, pipelineOpDelete, pipelineInfo.Input, pipelineInfo.Pipeline.Name); err != nil {
 			return nil, err
 		}
-		// delete the pipeline's output repo
-		if err := pachClient.DeleteRepo(request.Pipeline.Name, request.Force); err != nil {
-			return nil, err
+		if request.KeepRepo {
+			// Remove branch provenance (pass branch twice so that it continues to point
+			// at the same commit, but also pass empty provenance slice)
```

Comment:
Is this necessary to prevent further commit propagation into the ex-pipeline repo?

Reply:
Yeah, this is what signifies that you're "shutting off" the commit propagation.